Description: Fix bashims
Bug-Debian: https://bugs.debian.org/772246
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@users.sourceforge.net>
--- a/gnetlist/scripts/bom_xref.sh.in
+++ b/gnetlist/scripts/bom_xref.sh.in
@@ -3,7 +3,7 @@
 
 if [ -z "$1" ] ; then
     echo "usage $0 geda-bom  # output on stdout"
-    exit -1
+    exit 255
 fi    
 cat $1 | sort -k1,1 |\
 @AWK@ '!/device/{printf("%-5s %-20s %-20s %-20s\n", toupper($1), toupper($2), toupper($3), toupper($4));} /device/{}'
--- a/gnetlist/scripts/bompp.sh.in
+++ b/gnetlist/scripts/bompp.sh.in
@@ -3,7 +3,7 @@
 
 if [ -z "$1" ] ; then
     echo "usage $0 geda-bom  # output on stdout"
-    exit -1
+    exit 255
 fi    
 cat $1 | @AWK@ '{print toupper($2)" "toupper($3)" "toupper($4)" "toupper($1)}'\
 | sort +0 | @AWK@ '\
--- a/gnetlist/scripts/sch2eaglepos.sh
+++ b/gnetlist/scripts/sch2eaglepos.sh
@@ -17,7 +17,7 @@
 By Braddock Gaskill (braddock@braddock.com), August 2004
 
 EOF
-    exit -1
+    exit 255
 fi
 XOFFSET=40000
 YOFFSET=33000
@@ -44,11 +44,11 @@
 grep -B1 refdes= "$FNAME" |sed 's/=/ /' | cut -d" " -f2,3 |grep -v '^--' >${tmpf}
 
 
-while read; do
+while read REPLY; do
     # the directory on the client to backup
     X=`echo $REPLY | cut -d' ' -f1`
     Y=`echo $REPLY | cut -d' ' -f2`
-    read;
+    read REPLY;
     PART="$REPLY"
     X=`echo "scale=5; ($X - $XOFFSET) / $XSCALE" |bc`
     Y=`echo "scale=5; ($Y - $YOFFSET) / $YSCALE" |bc`
