Description: Fix possible security problems when displaying message string
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@sabily.org>
Bug: https://bugs.launchpad.net/geda/+bug/894400
Forwarded: https://bugs.launchpad.net/geda/+bug/894400/+attachment/2607078/+files/format-security.diff
--- a/gschem/src/gschem.c
+++ b/gschem/src/gschem.c
@@ -216,9 +216,9 @@
     GtkWidget* error_diag =
       gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_ERROR,
                               GTK_BUTTONS_OK,
-                              message);
+                              "%s", message);
     gtk_dialog_run (GTK_DIALOG (error_diag));
-    g_error (message);
+    g_error ("%s", message);
   }
 
   /* Allocate w_current */
