Description: enable build against Guile 2.2 in xorn
Author: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@users.sourceforge.net>
Bug: https://bugs.launchpad.net/geda/+bug/1782306
Index: geda-gaf/xorn/configure.ac
===================================================================
--- geda-gaf.orig/xorn/configure.ac	2019-10-19 08:30:42.666814571 +0200
+++ geda-gaf/xorn/configure.ac	2019-11-11 03:10:08.775094707 +0100
@@ -30,7 +30,7 @@
 
 AM_PATH_PYTHON([2.7])
 PKG_CHECK_MODULES([PYTHON], [python2 >= 2.7])
-PKG_CHECK_MODULES([GUILE], [guile-2.0 >= 2.0.10])
+PKG_CHECK_MODULES([GUILE], [guile-2.2 >= 2.2.0])
 
 AM_GNU_GETTEXT([external])
 AM_GNU_GETTEXT_VERSION([0.18.2])
